BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -669,25 +669,26 @@ PciSearchDevice (
 EFI_STATUS
 PciScanBus (
   IN PCI_IO_DEVICE                      *Bridge,
-  IN UINT8                              StartBusNumber,
-  OUT UINT8                             *SubBusNumber,
-  OUT UINT8                             *PaddedBusRange
+  IN UINT16                             StartBusNumber,
+  OUT UINT16                            *SubBusNumber,
+  OUT UINT16                            *PaddedBusRange

```

Comment:
UINT8 should be enough for bus number, do we really need change it to UINT16?

Reply:
This is because of "PrimaryBus <= PCI_MAX_BUS". PrimaryBus can be 256 and it causes UINT8 overflow, then 0 again.
This loop can be infinite with UINT8.
for (PrimaryBus = 0; PrimaryBus <= PCI_MAX_BUS; PrimaryBus = SubBusNumber + 1)

Reply:
Bus number range should be 0 ~ 255, totally at most 256 buses. How it could be 256?

Reply:
In PciEnumeration(),
for (PrimaryBus = 0; PrimaryBus <= PCI_MAX_BUS; PrimaryBus = SubBusNumber + 1) {
}
PrimaryBus should be 256(0x100) and then exit the for loop. But, UINT8 makes overflow.

Reply:
Got it.